src/monattack.cpp

```diff
@@ -1299,77 +1247,21 @@ void mattack::spit_sap(monster *z, int index)
         return;
     }
 
-    int t1, t2;
-    int dist;
     Creature *target = z->attack_target();
     if( target == nullptr ||
-        ( dist = rl_dist( z->pos(), target->pos() ) ) > 12 ||
-        !z->sees( *target, t1, t2 ) ) {
+        rl_dist( z->pos(), target->pos() ) > 12 ||
+        !z->sees( *target ) ) {
         return;
     }
 
-    auto msg_type = target == &g->u ? m_bad : m_neutral;
-
     z->moves -= 150;
     z->reset_special(index); // Reset timer
 
-    int deviation = rng(1, 10);
-    double missed_by = (.0325 * deviation * dist);
-    std::set<std::string> no_effects;
-
-    if (missed_by > 1.) {
-        if( g->u.sees( *z ) ) {
-            add_msg(_("The %s spits sap, but misses %s."), z->name().c_str(), target->disp_name().c_str() );
-        }
-
-        tripoint hitp( target->posx() + rng(0 - int(missed_by), int(missed_by)),
-                       target->posy() + rng(0 - int(missed_by), int(missed_by)),
-                       target->posz() );
-        std::vector<tripoint> line = line_to( z->pos(), hitp, 0, 0 );
-        int dam = 5;
-        for( auto &i : line ) {
-            g->m.shoot( i, dam, false, no_effects);
-            if (dam == 0 && g->u.sees( i )) {
-                add_msg(_("A glob of sap hits the %s!"),
-                        g->m.tername( i ).c_str());
-                return;
-            }
-
-            if (dam <= 0) {
-                break;
-            }
-        }
-
-        g->m.add_field( hitp, fd_sap, (dam >= 4 ? 3 : 2), 0 );
-        return;
-    }
-
-    if( g->u.sees( *z ) ) {
-        add_msg(_("The %s spits sap!"), z->name().c_str());
-    }
-    g->m.sees( z->pos(), target->pos(), 60, t1, t2 );
-    std::vector<tripoint> line = line_to( z->pos(), target->pos(), t1, t2 );
-    int dam = 5;
-    for (auto &i : line) {
-        g->m.shoot( i, dam, false, no_effects );
-        if (dam == 0 && g->u.sees( i )) {
-            add_msg(_("A glob of sap hits the %s!"),
-                    g->m.tername( i ).c_str());
-            return;
-        }
-    }
-    if (dam <= 0) {
-        return;
-    }
-    if( target->uncanny_dodge() ) {
-        return;
-    }
-    if( g->u.sees( *target ) ) {
-        add_msg( msg_type, _("A glob of sap hits %s!"), target->disp_name().c_str() );
-    }
-    target->deal_damage( z, bp_torso, damage_instance( DT_BASH, dam ) );
-    target->add_effect("sap", dam);
-    target->check_dead_state();
+    projectile proj;
+    proj.speed = 10;
+    proj.proj_effects.insert( "APPLY_SAP" );
```

Comment:
The flag, it does nothing. There is no other reference to it anywhere in the code or in JSON data.